Projects/Server/Gumps/New/Components/GridGumpBuilder.cs

```diff
@@ -0,0 +1,285 @@
+ *                                                                       *
+ * You should have received a copy of the GNU General Public License     *
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>. *
+ *************************************************************************/
+
+using Server.Gumps.Components.Interpolation;
+using Server.Gumps.Enums;
+using Server.Gumps.Interfaces;
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Text;
+using static Server.Gumps.Components.Interpolation.GumpInterpolatedStringHandler;
+
+namespace Server.Gumps.Components
+{
+    public ref struct GridGumpBuilder<T>
+        where T : struct, IStringsHandler
+    {
+        private GumpBuilder<T> _builder;
+
+        private readonly ushort _borderSize;
+        private readonly ushort _offsetSize;
+        private readonly ushort _entryHeight;
+        private readonly ushort _offsetGumpId;
+        private readonly ushort _headerGumpId;
+        private readonly ushort _entryGumpId;
+        private readonly ushort _backGumpId;
+        private readonly ushort _textHue;
+        private readonly ushort _textOffsetX;
+
+        private Span<byte> _backgroundSpan;
+        private Span<byte> _offsetSpan;
+        private int _currentX;
+        private int _currentY;
+        private int _backgroundWidth;
+        private int _offsetWidth;
+        private ushort _currentPage;
+
+        public readonly int CurrentX => _currentX;
+        public readonly int CurrentY => _currentY;
+        public readonly ReadOnlySpan<byte> Layout => _builder.Layout;
+        public readonly int LayoutSize => _builder.LayoutSize;
+        [UnscopedRef]
+        public ref T StringsWriter => ref _builder.StringsWriter;
+
+        public GridGumpBuilder(GumpFlags flags = GumpFlags.None, ushort borderSize = 10, ushort offsetSize = 1,
+            ushort entryHeight = 20, ushort offsetGumpId = 0x0A40, ushort headerGumpId = 0x0E14, ushort entryGumpId = 0x0BBC,
+            ushort backGumpId = 0x13BE, ushort textHue = 0, ushort textOffsetX = 2)
+        {
+            _builder = new(flags);
+
+            _borderSize = borderSize;
+            _offsetSize = offsetSize;
+            _entryHeight = entryHeight;
+            _offsetGumpId = offsetGumpId;
+            _headerGumpId = headerGumpId;
+            _entryGumpId = entryGumpId;
+            _backGumpId = backGumpId;
+            _textHue = textHue;
+            _textOffsetX = textOffsetX;
+        }
+
+        public void FinishPage()
+        {
+            if (_backgroundWidth == 0)
+            {
+                return;
+            }
+
+            int backgroundHeight = _currentY + _entryHeight + _offsetSize + _borderSize;
+            int offsetHeight = _currentY + _entryHeight + _offsetSize - _borderSize;
+
+            Span<char> buffer = stackalloc char[46 * 2];
```

Comment:
How did we come to the number 46 *2?
My estimate is that at most (which it never would be), it would be 8 characters per number, so 24 + 20 for the literals.
So 44 characters (no need to times it by 2 cause it is already chars?)